executive_logic/executive_logic.py

```diff
@@ -1,36 +1,204 @@
+from score_keeper.score_keeper import ScoreKeeper
+from ui.user_interface import UserInterface
+from wheel.wheel import Wheel, Sector
+from server import GameServer, QueryStatus, Message, MessageType
+from enum import Enum
+import time
+
 MAX_SPINS = 50
 
+
 class ExecutiveLogic:
-    def __init__(self):
-        self.num_spins = 1
-        self.round = 1
-        pass
+    def __init__(self, srv_ip, srv_port):
+        self.board = Board()  # Board object
+        self.score_keeper = ScoreKeeper()  # ScoreKeeper object
+        self.wheel = Wheel(["food", "politics"])  # Wheel object (TODO: These categories should be specified somewhere else, right?)
+        self.ui = UserInterface()  # UI Object
+        self.game_server = GameServer(srv_ip, srv_port, self)  # GameServer object
+
+        self.is_game_running = True  # True if game is ongoing, False otherwise
+        self.num_spins = 0  # Number of times wheel has been spun this round (round ends after MAX_SPINS)
+
+        self.query_status = QueryStatus.STANDBY
+        self.query_response = Message(MessageType.EMPTY, [])
+
+    def run_game(self):
+        """
+        Main game loop. Controls logic through the flow of the entire game.
+        :return:
+        """
+        self.is_game_running = True
+
+        for round_num in [1, 2]:
+            self.board.reset_board(round_num)
+            self.__execute_round(round_num)
+
+        self.__end_game()
+
+    def __execute_round(self, round_num):
+        """
+        Controls the flow of one round.
+        :param round_num: Number of current round (1 or 2).
+        :return: void
+        """
+        self.num_spins = 0
+        curr_player_id = None
+        # TODO: At start of round, send Jeopardy board to server (then client) with UPDATE_BOARD MessageType
+
+        while self.num_spins < MAX_SPINS and self.board.get_available_categories(
+                round_num):  # End round when spins >= 50 or no available questions
+            curr_player_id = self.__next_player(curr_player_id)
+            self.__query_server(MessageType.SPIN, [curr_player_id])  # Notify player that it's their turn,  ask them to push a button to spin the wheel, and wait for their response
+            self.__execute_turn(curr_player_id, round_num)
+
+    def __execute_turn(self, curr_player_id, round_num):
+        """
+        Controls the flow of one turn. Called at the start of the turn.
+        :param curr_player_id: ID of player whose turn is starting
+        :return: void
+        """
+
+        # Spin wheel
+        if self.num_spins >= MAX_SPINS:
+            return
+        wheel_result = self.__spin_wheel()
+        #self.__query_server(MessageType.SPIN_RESULT, [wheel_result])  # TODO: Implement, so that user can see what sector they spun
+
+        # If result is a Jeopardy category, call execute_category
+        if self.wheel.is_jeopardy_category(wheel_result):
+            self.__execute_category(wheel_result, curr_player_id, round_num)
```

Comment:
This should continue if the player gets it right. I think you need to return `is_correct` from __execute_category and continue until the category is empty or the player `is_correct` is False.